### PR TITLE
5 add main page

### PR DESCRIPTION
**Pull request by Dergacz**

### CODE REVIEW

File: .eslintrc.yml [hunk 1]
```diff
@@ -2,14 +2,14 @@ env:
...
 parserOptions:
   ecmaVersion: latest
+  project: ["tsconfig.json"]
 plugins:
   - react
   - '@typescript-eslint'
-rules: {
-  indent: ['space', 2]
-}
+  - 'import'
```
Comment (melonholy) [inline .eslintrc.yml:15]: new line

File: app.tsx [hunk 2]
```diff
@@ -1,7 +1,9 @@
 import * as React from 'react';
+import MainPage from './src/components/MainPage';
+
 function App() {
-    return (
-        <div>{'react setup from scratch without cra testing ts'}</div>
-    );
+  return (
+        <MainPage/>
```
Comment (melonholy) [inline app.tsx:6]: did you applied prettier seems like you have 4 spaces on tab instead of 2?

File: src/components/MainPage.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import * as React from 'react';
+import { pokemonApi } from '../services/PokemonServiceApi';
+import { Pokemon } from './Pokemon';
+import { FC } from 'react';
```
Comment (melonholy) [inline src/components/MainPage.tsx:4]: Don't duplicate react import

File: src/components/MainPage.tsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+const MainPage: FC = () => {
+  const { data: pokemons } = pokemonApi.useFetchAllPokemonQuery(100);
+  return (
+        <div>
+            {
+                pokemons && pokemons.results.map(pokemon => (
```
Comment (melonholy) [inline src/components/MainPage.tsx:11]: it's better to use pokemons?.results.map instead of check

File: src/components/MainPage.tsx [hunk 5]
```diff
@@ -0,0 +1,23 @@
+import * as React from 'react';
+import { pokemonApi } from '../services/PokemonServiceApi';
+import { Pokemon } from './Pokemon';
+import { FC } from 'react';
+
+const MainPage: FC = () => {
+  const { data: pokemons } = pokemonApi.useFetchAllPokemonQuery(100);
+  return (
+        <div>
```
Comment (melonholy) [inline src/components/MainPage.tsx:9]: Maybe you should move it in separate component like PokemonsList

File: src/components/Pokemon.tsx [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+
+export const Pokemon: FC<IPokemon> = ({
+  name,
+}) => {
+  return (
+        <div>
+            {name}
+        </div>
+  );
+};
```
Comment (melonholy) [inline src/components/Pokemon.tsx:13]: new line

File: src/models/models.ts [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+export interface IPokemon {
+  name: string;
+  url: string;
+}
+export interface IFetchPokemons {
+  count: number;
+  next: string | null;
+  previous: string | null;
+  results: IPokemon[];
+}
```
Comment (melonholy) [inline src/models/models.ts:11]: new line

File: src/models/models.ts [hunk 8]
```diff
@@ -0,0 +1,11 @@
+
```
Comment (melonholy) [inline src/models/models.ts:1]: i think it's better to have file named [ComponentFolderName].types.ts in component folder
and lets make folder structure for components smth like this
image

File: src/services/PokemonServiceApi.ts [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+    fetchAllPokemon: build.query<IFetchPokemons, number>({
+      query: (limit: number = 10) => ({
+        url: 'pokemon',
+        params: {
+          limit,
+        },
+      }),
+    }),
+  }),
+});
```
Comment (melonholy) [inline src/services/PokemonServiceApi.ts:17]: new line

**Review by melonholy — CHANGES_REQUESTED** [review]

File: src/components/MainPage.tsx (hunk 4 above)
Comment (Dergacz) [reply src/components/MainPage.tsx]: the request takes time to pass, so the pokemons will be undefined for a while

Comment (melonholy) [reply src/components/MainPage.tsx]: yes, but optional chaining will prevent further operations

**Pull request merged**